### PR TITLE
feat(input): input / label error self management

### PR DESCRIPTION
**Pull request by okkindel**
- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Feature

## Which package are you modifying?

- [x] input

## What is the current behavior?

I was wondering about integrating Input with Reactive Forms when it comes to presenting the error state - in my opinion, since the library is being created for Angular, it is worth considering. I am presenting the initial concept, you can safely ignore it if you don't like it, it is basically not so much a ready PR as a POC solution - maybe it's just worth discussing. See if you see such a solution in your project.

## What is the new behavior?

Input for the `error` state and integration with NgControl which allows for self-management of the components error state.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by okkindel** [comment]
Actually, classes are a better idea, I don't know why I didn't think of it - we get rid of a whole lot of code. I like this much more, I would make one change, with errorAware we can only control whether the control should react to `its' errors, but we cannot force this error from the outside, which may be useful for people who do not use ReactiveForms or want to make the state dependent from other variables, so I would introduce a slightly different implementation:

```
error: {
  auto: '[.ng-invalid]:text-destructive [.ng-invalid]:border-destructive [.ng-invalid]:focus-visible:ring-destructive',
  true: 'text-destructive border-destructive focus-visible:ring-destructive',
},

....

defaultVariants: {
  [...]
  error: 'auto',
},
```

Additionally, we should react to ng-invalid and ng-touched at the same time, because we don't want to see an error before we start entering data on an empty form with required controls.

However, I don't know how to create these conditional statements, because the method you provided will not work, this way of writing conditions works for groups: https://tailwindcss.com/docs/hover-focus-and-other-states#arbitrary-groups, but not for `host`.

**Comment by okkindel** [comment]
I've updated the code - let me know what you think.

**Comment by goetzrobin** [comment]
I like it! I think this is absolutely going into the right direction.
This might be another PR, but I think we could use the same approach to the label.
And add something like this:
```
[&:has([hlmInput].ng-invalid.ng-touched)]:text-destructive [&:has([hlmInput].ng-invalid.ng-touched)]:text-destructive
```

One thing, could you add a story that uses ngModel?

**Comment by goetzrobin** [comment]
We might also want to add a "false" variant doing nothing

**Comment by okkindel** [comment]
We have false by default: 
image

**Comment by okkindel** [comment]
I added changes to the storybook

**Comment by okkindel** [comment]
I also added variant for labels - in a similar style

File: libs/ui/input/helm/src/lib/hlm-input.directive.ts [hunk 1]
```diff
@@ -12,9 +12,14 @@ const inputVariants = cva(
         sm: 'h-9 px-3',
         lg: 'h-11 px-8',
       },
+      error: {
+        auto: '[&.ng-invalid&.ng-touched]:text-destructive [&.ng-invalid&.ng-touched]:border-destructive [&.ng-invalid&.ng-touched]:focus-visible:ring-destructive',
```
Comment (goetzrobin) [inline libs/ui/input/helm/src/lib/hlm-input.directive.ts:16]:
```suggestion
        auto: '[&.ng-invalid.ng-touched]:text-destructive [&.ng-invalid.ng-touched]:border-destructive [&.ng-invalid.ng-touched]:focus-visible:ring-destructive',
```
Same here

**Review by goetzrobin — CHANGES_REQUESTED** [review]
Two minor changes! Rest looks great! Thanks for adding this!

File: libs/nx/src/generators/ui/libs/ui-input-helm/files/lib/hlm-input.directive.ts.template [hunk 2]
```diff
@@ -12,9 +12,14 @@ const inputVariants = cva(
         sm: 'h-9 px-3',
         lg: 'h-11 px-8',
       },
+      error: {
+        auto: '[&.ng-invalid&.ng-touched]:text-destructive [&.ng-invalid&.ng-touched]:border-destructive [&.ng-invalid&.ng-touched]:focus-visible:ring-destructive',
```
[the remark this answers is not in the document]
Comment (okkindel) [reply libs/nx/src/generators/ui/libs/ui-input-helm/files/lib/hlm-input.directive.ts.template:16]: It is IMO, i was testing it, we are catching `host` classes.

**Review by goetzrobin — APPROVED** [review]

**Pull request merged**